### DISCUSSION TITLE
ASIN Listing products

### DISCUSSION THREAD

**Discussion by Rudy9419** [amzn/selling-partner-api-models#3105] [General]
Hey Guys,

I was looking in the last few days for a way/best practice to list products that already exist in Amazon. Can anybody help me chosing the correct APIs to use? For now the only solution I found would be getting the ASIN parameters from CatalogAPI and create a product with that parameters and my sku. But this doesn't seem like a best practice.

I am open to all suggestions.

**Comment by teddy-codes** [comment]
You would basically have to do just that. With Amazon, there can be multiple ASINs attached to one UPC and you would just have to figure out the best one that is actually the product that you have on hand. It's definitely a touchy situation when you try to automate this. There even could be a scenario where the UPC points to the correct product, but it's listed as a two pack. I would be careful if you're automating this.

**Reply by Rudy9419** [reply]
> There even could be a scenario where the UPC points to the correct product, but it's listed as a two pack

I get this error a lot, not really with 2 pack but similar on itemPacageQuantity... Do you have any solution for this?

And another problem is like I am trying to list the good data and I get that something exist with the ASIN BXXXX and I should corect my data to make it be the same, but my data is correct. How to solve this, like I would like to create the item with the data from ASIN if Amazon sais that is the good one, I don't mind but he doesn't want to create on ASIN as well... It is really frustrating.

>  I would be careful if you're automating this.

Unfortunatly I kinda need this automated, but when I provide ASIN on StandardProductID and EAN on RelatedProductID I get an error that EAN differ from amazon and the ean I provide is null... Do you know where I should provide the ean if not in RelatedProductID?

**Reply by teddy-codes** [reply]
> I get this error a lot, not really with 2 pack but similar on itemPacageQuantity... Do you have any solution for this?

No, not really. This is something that is a problem with Amazon. They have garbage listing validation (basically doesn't exist), so it's in part their fault. They allow sellers to make a bunch of listings for the same UPC which causes this problem here. Sellers, of course, don't want to compete with other sellers. If that means creating a variation for a two pack on a listing that was already a two pack or creating another listing with the same UPC, etc.. No one wants to compete with each other for the buy box on any product. Amazon doesn't account for this. My thoughts on that are to train an AI model to tell you what the package quantity is, but it could potentially be wrong. You would have to get a data set together that allows you to train the model, etc. That's a huge task.

**Comment by MST1122** [comment]
@Rudy9419 I have a question, how are you generating UI for user input for listing product data?

**Reply by Rudy9419** [reply]
Hey @MST1122 , I don't generate UI for user.

**Reply by MST1122** [reply]
then how are you preparing data for upload?

**Reply by Rudy9419** [reply]
This is kinda confidential, but I can say that we are working with files and we get all the required data from the files.

**Reply by MST1122** [reply]
please can you give me an idea, what kind of files you are talking about? I have been wandering around for more than 5 weeks and could not find a proper solution for this?

**Reply by Rudy9419** [reply]
I don't really know what kind of information you need. Please get me up to date with the issue you are facing and maybe I can help you.

**Reply by MST1122** [reply]
I need to do product listing with the new LISTING API (and it's my first experience with Amazon API). I am able to get product type definitions in JSON-schema form, next I need to generate a User Interface based on that schema for a user to input data.

**Reply by Rudy9419** [reply]
I am using the old API... Do you have access to the sellercentral on Amazon? Maybe you can do something similar something with tabs where it should have grouped properties like variation specific, general and product type specific. But how I mentioned I am using the POST_PRODUCT_DATA with XML request not json and I have the XSD for categories and all. In my files I get all the required attributes to publish new prodcuts on amazon. 
Maybe this [link](https://gist.github.com/kitwalker12/87a6939540a948cac7041459aa718189) will help you.

**Reply by littleylv** [reply]
@MST1122 Hi, have you found a solution to "generate a User Interface based on that schema for a user to input data"

**Reply by chapmanjw** [reply]
This works fairly well with the JSON Schemas provided by the Product Type Definitions API: https://rjsf-team.github.io/react-jsonschema-form/. (When using their online demo, you have to remove the `"$schema"` property from the JSON Schema document).

**Comment by ilguala** [comment]
How can i pass ASIN parameter in Put Listing Items API? in the guide there is not Asin property to tell that the item i am loading is related to that amazon Asin. In xsd feed schema the property is StandarProductId. Is there an equivalent?

**Reply by vaidas-lungis** [reply]
There is a `merchant_suggested_asin` attribute 
Example https://github.com/amzn/selling-partner-api-docs/blob/0b6dc69fefd9c1d85ed7ebadd4d2d1af55eff305/guides/en-US/use-case-guides/listings-items-api-use-case-guide/listings-items-api-use-case-guide_2021-08-01.md#L252

![Screenshot 2021-11-30 at 13 10 54](https://user-images.githubusercontent.com/5111555/144036918-2108e501-c0d0-4e67-ad2b-d53577c4cc66.png) 

How you tried use that attribute?

**Reply by ilguala** [reply]
Ok, but if i submit request similar to yours, Amazon api returns many required attributes errors. 
It seems like that Amazon Api consider my request as a not existing product in catalog.  Instead the EAN/ASIN is already registered in Amazon Catalog. If i submit product request with xsd feed all works well, no extra attributes are required.
